File: LogicBuildingProblems/largeNumberDiv11.go
package main

import "fmt"

func checkDivisible11(str string) bool {
	size := len(str)
	bytes := []byte(str)
	sumOdd, sumEvent := 0, 0
	for i := 0; i < size; i++ {
		if i%2 == 0 {
			sumEvent += int(bytes[i] - byte('0'))
		} else {
			sumOdd += int(bytes[i] - byte('0'))
		}
	}
	if (sumEvent-sumOdd)%11 == 0 {
		return true
	}
	return false
}
func main() {
	var str string
	str = "7695"
	fmt.Println(checkDivisible11(str))
}

File: LogicBuildingProblems/largeNumberDiv11.cpp
#include<iostream>
#include<string>
using namespace std;
bool checkDivisible11(string str){
    int n = str.size();
    int sumOdd{}, sumEvent{};
    for (int i = 0; i < n; ++i){
        if(i & 1){
            sumEvent += str[i] - '0';
        }else{
            sumOdd += str[i] - '0';
        }
    }
    return (sumEvent - sumOdd) % 11 == 0;
}
int main(){
    string s = "76945";
    cout << (checkDivisible11(s) ? "true" : "false");

}